controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -366,10 +367,13 @@ func (r *StorageClassClaimReconciler) reconcileProviderPhases() (reconcile.Resul
 	r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimInitializing
 
 	// reading storageConsumer Name from storageClassClaim ownerReferences
+	r.storageConsumer = &v1alpha1.StorageConsumer{}
 	ownerRefs := r.storageClassClaim.GetOwnerReferences()
 	for i := range ownerRefs {
-		if ownerRefs[i].Kind == r.storageConsumer.Kind {
+		if ownerRefs[i].Kind == "StorageConsumer" {
 			r.storageConsumer.Name = ownerRefs[i].Name
+			r.storageConsumer.Namespace = r.storageClassClaim.Namespace
```

Comment:
Don't assume same namespace for Claim (user accesiable API)  and consumer (internal API), use storage clsuter instead (which is already loaded):
```suggestion
			r.storageConsumer = &v1alpha1.StorageConsumer{}
			r.storageConsumer.Name = ownerRefs[i].Name
			r.storageConsumer.Namespace = r.storageCluster.Namespace
```